src/Core/Components/DataGrid/FluentDataGridCell.razor.cs

```diff
@@ -61,4 +61,29 @@ protected override void OnInitialized()
 
     public void Dispose() => Owner.Unregister(this);
 
+    private bool StopClickPropagation()
+    {
+        var selColumn = Owner.Owner.Grid.SelectColumns.FirstOrDefault();
```

Comment:
You need to manage all `SelectColumns`, not just the first one. It is possible to have several `SelectColumns` in the DataGrid.